### ISSUE TITLE
Feature Request: Was to intercept sent chat messages client side

### ISSUE THREAD

**Issue by Kilobyte22** [MinecraftForge/MinecraftForge#1442]
So, i want a way to intercept sent chat messages without the server receiving them. Normally i could use a coremod, but i think since this could be useful to other people too, it might be better to be added into forge. The easiest way would probably to make a (cancallable) `ClientChatSentEvent`. To archieve the actual call, `func_146403_a()` in GuiChat should be changed to:

``` java
this.mc.ingameGUI.getChatGUI().addToSentMessages(p_146403_1_);
ClientChatSentEvent event = new ClientChatSentEvent(p_146403_1_);
if (!MinecraftForge.EVENT_BUS.post(event) && event.message != null) {
    if (net.minecraftforge.client.ClientCommandHandler.instance.executeCommand(mc.thePlayer, p_146403_1_) == 1) return;
    this.mc.thePlayer.sendChatMessage(p_146403_1_);
}
```

Edit: While at it, it might be wise to also provide a way to intercept tab-completion. I haven't looked at the code for that yet though

**Comment by EZtouch** [comment]
We have ClientChatReceivedEvent that allows us to manipulate and monitor the chat that the client receives, ServerChatEvent to monitor the chat that the server receives, but we have no way to handle the chat that the player is about to send if we are considering a client-only mod. I need this for my chat translating mod, and the only way I can think of to manipulate the sent chat messages is to modify the built-in vanilla chat and make my own just for storing the user's chat input and manipulating it, which i find impractical.

**Comment by MrCreeperkopf** [comment]
Is it now included into Forge 1.8?

**Comment by mezz** [comment]
This event seems to have a reasonable case for it. 
To take this to the next stage, someone should write a PR.
